### PR TITLE
configure: fix libsodium issues

### LINKED ISSUE #3171: build failed: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’
### Issue and Steps to 
```
git checkout v0.7.3rc2
make clean && make distclean
./configure
make

....

hsmd/hsmd.c: In function ‘create_encrypted_hsm’:
hsmd/hsmd.c:498:2: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’
  crypto_secretstream_xchacha20poly1305_state crypto_state;
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:499:12: error: ‘crypto_secretstream_xchacha20poly1305_HEADERBYTES’ undeclared (first use in this function)
  u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:499:12: note: each undeclared identifier is reported only once for each function it appears in
hsmd/hsmd.c:501:36: error: ‘crypto_secretstream_xchacha20poly1305_ABYTES’ undeclared (first use in this function)
  u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:503:2: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_init_push’ [-Werror=implicit-function-declaration]
  crypto_secretstream_xchacha20poly1305_init_push(&crypto_state, header,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:505:2: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_push’ [-Werror=implicit-function-declaration]
  crypto_secretstream_xchacha20poly1305_push(&crypto_state, cipher,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:501:5: error: unused variable ‘cipher’ [-Werror=unused-variable]
  u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
     ^~~~~~
hsmd/hsmd.c:499:5: error: unused variable ‘header’ [-Werror=unused-variable]
  u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
     ^~~~~~
hsmd/hsmd.c: In function ‘load_hsm’:
hsmd/hsmd.c:636:3: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’
   crypto_secretstream_xchacha20poly1305_state crypto_state;
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:637:13: error: ‘crypto_secretstream_xchacha20poly1305_HEADERBYTES’ undeclared (first use in this function)
   u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:639:37: error: ‘crypto_secretstream_xchacha20poly1305_ABYTES’ undeclared (first use in this function)
   u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:647:7: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_init_pull’ [-Werror=implicit-function-declaration]
   if (crypto_secretstream_xchacha20poly1305_init_pull(&crypto_state, header,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:651:7: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_pull’ [-Werror=implicit-function-declaration]
   if (crypto_secretstream_xchacha20poly1305_pull(&crypto_state,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:639:6: error: unused variable ‘cipher’ [-Werror=unused-variable]
   u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
      ^~~~~~
hsmd/hsmd.c:637:6: error: unused variable ‘header’ [-Werror=unused-variable]
   u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
      ^~~~~~
cc1: all warnings being treated as errors
<builtin>: recipe for target 'hsmd/hsmd.o' failed
make: *** [hsmd/hsmd.o] Error 1

```

### PR DESCRIPTION
**Pull request by darosior**
fix #3171
hacked-by: @Saibato

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -241,6 +241,7 @@ code=
 
 int main(void)
 {
+	crypto_secretstream_xchacha20poly1305_state crypto_state;
```
Comment (rustyrussell) [inline configure]: This doesn't work, because gcc gives an unused variable warning!

I took the actual code we use in the hsm, and simplified it a little.

**Comment by darosior** [comment]
ACK 5dcd750c5afc6b8c988d14bafa4f56e95632c2f5

Tested on a machine on which the build didn't pass.
Squashed the fixup (Don't know how to keep author though..)

**Comment by Saibato** [comment]
> ACK [5dcd750](https://github.com/ElementsProject/lightning/commit/5dcd750c5afc6b8c988d14bafa4f56e95632c2f5)
> 
> Tested on a machine on which the build didn't pass.
> Squashed the fixup (Don't know how to keep author though..

Yes that's always tough, compile always our imported libsodium or trust the installed one.
The space between `state and crypto_state` is small. I would prefer to build all in source.
What was more quick fix twinkle, is now in discussion. Can't we try to define a configure override flag  to
always compile sodium from our source or not ?

**Comment by darosior** [comment]
> What was more quick fix twinkle, is now in discussion. Can't we try to define a configure override flag to
always compile sodium from our source or not ?

I think this should be for 0.7.4 as we're about to release

**Pull request merged**